### PR TITLE
1947 decouple programs module from dispensary mode check

### LINKED ISSUE #1947: Decouple `Programs` module from `Dispensary` mode check
## Is your feature request related to a problem? Please describe 👀
Need a way to hide programs when dispensary mode is on. 

## Describe the solution you'd like 🎁
Programs should be hidden behind a `preference`, so that the links do not show up when a store is in `dispensary mode`. This can be done as a `store preference` and will require changes in mSupply. All program related components and columns will also need to be hidden from `Patient` list view.

### Describe alternatives you've considered 💭
Keep the `Program` navigation links since they will be empty if the `Programs` module isn't configured, but seems weird to have links that lead to empty pages...

### Moneyworks Jobcode 🧰

OMS:MVP | Open mSupply MVP Prescription Module
-- | --

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #1947

# 👩🏻‍💻 What does this PR do? 
Hides all program related components with store preference. 

# 🧪 How has/should this change been tested? 
Will need [this branch](https://github.com/openmsupply/msupply/tree/%2313221-omSupply-store-preference-for-Programs-module) for testing.
Make sure all program related things are hidden if omSupply: uses Program module is not configured for store.

### CODE REVIEW

File: client/packages/system/src/Patient/PatientView/AppBarButtons.tsx [hunk 1]
```diff
@@ -28,6 +32,7 @@ export const AppBarButtons: FC<{ disabled: boolean }> = ({ disabled }) => {
     },
   });
   const disableEncounterButton = enrolmentData?.nodes?.length === 0;
+  if (!store?.preferences.omProgramModule) return null;
```
Comment (mark-prins) [inline client/packages/system/src/Patient/PatientView/AppBarButtons.tsx:35]: I would expect the `Print` button to still be available for non-program patients - shouldn't it?

File: client/packages/system/src/Patient/PatientView/PatientView.tsx [hunk 2]
```diff
@@ -206,13 +208,17 @@ export const PatientView = () => {
           }}
         />
       ) : null}
-      <AppBarButtons disabled={!!createNewPatient} />
+      <AppBarButtons disabled={!!createNewPatient} store={store} />
       <PatientSummary />
-      {/* Only show tabs for saved patients */}
+      {/* Only show tabs if program module is on and patient is saved.
```
Comment (mark-prins) [inline client/packages/system/src/Patient/PatientView/PatientView.tsx:213]: I was going to complain about the multiple `?` operators and refactored to the following, then realised that this is just what the comment says it should be..

```
      {!createNewPatient && store?.preferences.omProgramModule ? (
        <DetailTabs tabs={tabs} requiresConfirmation={requiresConfirmation} />
      ) : (
        <PatientDetailView onEdit={setIsDirtyPatient} />
      )}
```

**Review by mark-prins — CHANGES_REQUESTED** [review]
🎉  That's working nicely

I wonder if the `Reports` page should still be shown though - that isn't program specific.
We do have a problem if some stores are using programs and have program specific reports I guess. that's an issue because reports aren't store specific. perhaps they need to be?

I also wonder if the `NUIC` should be hard coded as the default for everyone? NUIC is a specific term

Screenshot 2023-07-18 at 4 16 48 PM

**Comment by roxy-dao** [comment]
> I wonder if the `Reports` page should still be shown though - that isn't program specific. We do have a problem if some stores are using programs and have program specific reports I guess. that's an issue because reports aren't store specific. perhaps they need to be?
> 
> I also wonder if the `NUIC` should be hard coded as the default for everyone? NUIC is a specific term

Reports is currently program specific... There was talks about moving it out of Dispensary and re-designing UI to have other reports as well, but thought to hide it for now until that is done...

Can remove NUIC if isn't applicable for everyone (:

File: client/packages/system/src/Patient/PatientView/PatientView.tsx (hunk 2 above)
Comment (roxy-dao) [reply client/packages/system/src/Patient/PatientView/PatientView.tsx:213]: I just re-read it 😅 
[ad327d8](https://github.com/openmsupply/open-msupply/pull/1998/commits/ad327d846bb2d0d7e95a94d30e8df45fadcdb5fe)

**Review by mark-prins — APPROVED** [review]
👍  thanks for changes - I think we leave the patient search form question for another day
The reports too.. that's a wider problem than this issue covers

**Pull request merged**